SU2_CFD/src/numerics_direct_mean.cpp

```diff
@@ -2613,1427 +2348,954 @@ if (sM > 0.0) {
+  
+  su2double nVel[3];
+  nVel[0] = vel[0]*nV[0];
+  nVel[1] = vel[1]*nV[1];
+  nVel[2] = vel[2]*nV[2];
+  su2double U_k = nVel[0]+nVel[1]+nVel[2];
+  su2double vSquHlf = 0.5*vecDotVec3d(vel, vel);
+  su2double c = sqrt(gamma*pp/rrho);
+  su2double inv_kap_m1 = 1.0/kapm1;
+  
+  A[0][0] =-surfVeloc;
+  A[0][1] = nV[0];
+  A[0][2] = nV[1];
+  A[0][3] = nV[2];
+  A[0][4] = 0.0;
+  
+  A[1][0] = -vel[0]*(nVel[1]+nVel[2])+nV[0]*(kapm1*vSquHlf-vel[0]*vel[0]);
+  A[1][1] = (2.-gamma)*nVel[0]+U_k-surfVeloc;
+  A[1][2] = vel[0]*nV[1]-kapm1*vel[1]*nV[0];
+  A[1][3] = vel[0]*nV[2]-kapm1*vel[2]*nV[0];
+  A[1][4] = kapm1*nV[0];
+  
+  A[2][0] = -vel[1]*(nVel[0]+nVel[2])+nV[1]*(kapm1*vSquHlf-vel[1]*vel[1]);
+  A[2][1] = -kapm1*vel[0]*nV[1]+ vel[1]*nV[0];
+  A[2][2] = (2.-gamma)*nVel[1]+U_k-surfVeloc;
+  A[2][3] = vel[1]*nV[2]-kapm1*vel[2]*nV[1];
+  A[2][4] = kapm1*nV[1];
+  
+  A[3][0] = -vel[2]*(nVel[0]+nVel[1])+nV[2]*(kapm1*vSquHlf-vel[2]*vel[2]);
+  A[3][1] = -kapm1*vel[0]*nV[2]+vel[2]*nV[0];
+  A[3][2] = -kapm1*vel[1]*nV[2]+vel[2]*nV[1];
+  A[3][3] = (2.-gamma)*nVel[2]+U_k-surfVeloc;
+  A[3][4] = kapm1*nV[2];
+  
+  A[4][0] = U_k*((gamma-2.)*vSquHlf-c*c*inv_kap_m1);
+  A[4][1] = c*c*inv_kap_m1*nV[0]-kapm1*vel[0]*(nVel[1]+nVel[2])-(kapm1*vel[0]*vel[0]-vSquHlf)*nV[0];
+  A[4][2] = c*c*inv_kap_m1*nV[1]-kapm1*vel[1]*(nVel[0]+nVel[2])-(kapm1*vel[1]*vel[1]-vSquHlf)*nV[1];
+  A[4][3] = c*c*inv_kap_m1*nV[2]-kapm1*vel[2]*(nVel[0]+nVel[1])-(kapm1*vel[2]*vel[2]-vSquHlf)*nV[2];
+  A[4][4] = gamma*U_k-surfVeloc;
+  
+}
 
-//  RoeKappaStaticEnthalpy = 0.5*(StaticEnthalpy_i*Kappa_i + StaticEnthalpy_j*Kappa_j);
-//  RoeKappaStaticEnthalpy = (StaticEnthalpy_i*Kappa_i + StaticEnthalpy_j*Kappa_j + 4*RoeKappaStaticEnthalpy)/6;
-//  s = RoeChi + RoeKappaStaticEnthalpy;
-//  D = s*s*delta_rho*delta_rho + delta_p*delta_p;
-//  delta_rhoStaticEnergy = Density_j*StaticEnergy_j - Density_i*StaticEnergy_i;
-//  err_P = delta_p - RoeChi*delta_rho - RoeKappa*delta_rhoStaticEnergy;
-//
-//
-//  if (abs((D - delta_p*err_P)/Density_i)>1e-3 && abs(delta_rho/Density_i)>1e-3 && s/Density_i > 1e-3) {
-//
-//    RoeKappa = (D*RoeKappa)/(D - delta_p*err_P);
-//    RoeChi = (D*RoeChi+ s*s*delta_rho*err_P)/(D - delta_p*err_P);
-//
-//  }
 
-  RoeSoundSpeed2 = RoeChi + RoeKappa*(RoeEnthalpy-0.5*sq_vel);
+#endif
 
-}
 
-CUpwL2Roe_Flow::CUpwL2Roe_Flow(unsigned short val_nDim, unsigned short val_nVar, CConfig *config) : CNumerics(val_nDim, val_nVar, config) {
-    
+CUpwRoeBase_Flow::CUpwRoeBase_Flow(unsigned short val_nDim, unsigned short val_nVar, CConfig *config,
+                                   bool val_low_dissipation) : CNumerics(val_nDim, val_nVar, config) {
+  
   implicit = (config->GetKind_TimeIntScheme_Flow() == EULER_IMPLICIT);
   grid_movement = config->GetGrid_Movement();
   kappa = config->GetRoe_Kappa(); // 1 is unstable
-  
+
   Gamma = config->GetGamma();
   Gamma_Minus_One = Gamma - 1.0;
   
+  roe_low_dissipation = val_low_dissipation;
+  
   Diff_U = new su2doub
```

Comment:
The Bogeyman came for the Roe classes. Roe, L2Roe, and LMRoe now share a parent that implements the `fc_{1/2} = kappa*(fc_i+fc_j)*Normal` part, the rest is then done by the derived classes. And with this I concluded this type of refactoring.